pytorch_to_returnn/torch/_C.py

```diff
@@ -122,17 +123,48 @@ class Size(tuple):  # type: Tuple[SizeValue, ...]
 
 class SizeValue(int):
   """
-  We extend this, to store extra information, e.g. such that this reflects the batch-dim.
+  We extend this, to store extra information, e.g. corresponding RETURNN dim tags.
   """
-  is_batch_dim: bool = False
-  merged_dims: list[SizeValue] = []
+  def __new__(cls, x: int, dim_tag: Optional[Dim] = None, merged_dims: Optional[List[SizeValue]] = None, tensor=None, axis=None):
+    res = super(SizeValue, cls).__new__(cls, x)
+    res.dim_tag = dim_tag or Dim(dimension=x, description="static_dim")
+    res.merged_dims = merged_dims or []
+    res.tensor = tensor
+    res.axis = axis
+    return res
+
+  @property
+  def is_batch_dim(self):
+    if self.dim_tag is None:
+      return False
+    return self.dim_tag.is_batch_dim()
 
   def __repr__(self):
     res = super(SizeValue, self).__repr__()
     if self.is_batch_dim:
       res = f"Batch({res})"
     return res
 
+  def __mul__(self, other):
+    assert isinstance(other, (int, SizeValue)), (
+      "Multiplying a SizeValue with object of type {} is not allowed because it can lead to bugs, e.g. assumtion of a "
+      "static batch dim.".format(type(other)))
```

Comment:
Note that we might want to allow this for static dims. But not so important now.

Reply:
I'll add a comment.